analytics-core/src/main/java/com/segment/analytics/SegmentDispatcher.java

```diff
@@ -251,7 +251,12 @@ void performFlush() {
     try {
       int payloadsUploaded = networkExecutor.submit(new Callable<Integer>() {
         @Override public Integer call() throws Exception {
-          return upload();
+            try {
+                return upload();
+            } catch(IOException ioe) {
+                error(ioe, "Upload failed: " + ioe.getMessage());
+                return -1;
```

Comment:
I removed that try / catch block entirely and went back to the original code.
I also cleared the application data (by uninstalling the test app).
Although the StackOverflowExceptionis now gone  I can still reproduce the EOFException given the steps provided

As you can see it is now wrapped inside an IOError
```
05-20 15:45:37.541    6281-6303/? E/AndroidRuntime﹕ FATAL EXCEPTION: SegmentAnalytics-SegmentDispatcher
    java.io.IOError: java.io.EOFException
            at com.segment.analytics.SegmentDispatcher.performEnqueue(SegmentDispatcher.java:177)
            at com.segment.analytics.SegmentDispatcher$SegmentDispatcherHandler.handleMessage(SegmentDispatcher.java:405)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:137)
            at android.os.HandlerThread.run(HandlerThread.java:61)
     Caused by: java.io.EOFException
            at java.io.RandomAccessFile.readFully(RandomAccessFile.java:419)
            at com.segment.analytics.QueueFile.ringRead(QueueFile.java:255)
            at com.segment.analytics.QueueFile.remove(QueueFile.java:505)
            at com.segment.analytics.QueueFile.remove(QueueFile.java:477)
            at com.segment.analytics.SegmentDispatcher.performEnqueue(SegmentDispatcher.java:175)
            at com.segment.analytics.SegmentDispatcher$SegmentDispatcherHandler.handleMessage(SegmentDispatcher.java:405)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:137)
            at android.os.HandlerThread.run(HandlerThread.java:61)

```